source/blog/2017-08-16-ovirt-ansible-roles-how-to-use.html.md

```diff
@@ -0,0 +1,149 @@
+
+```bash
+$ yum install -y http://resources.ovirt.org/pub/yum-repo/ovirt-release41.rpm
+```
+
+oVirt release repository now provides `ovirt-ansible-roles` package:
+
+```bash
+$ yum install -y ovirt-ansible-roles
+```
+
+The structure of `ovirt-ansible-roles` package is following:
+
+ - __/usr/share/ansible/roles__ - stores the roles
+ - __/usr/share/doc/ovirt-ansible-roles/__ - stores the examples, basic overview and licence
+ - __/usr/share/doc/ansible/roles/{role_name}__ - store the documentation specific for role
+
+Now copy one of the examples from directory `/usr/share/doc/ovirt-ansible-roles/examples/` into your working directory modify the needed variables and run the playbook.
+
+## Example of tweaking and running the playbook
+
+## Create example playbook
+
+The proccess of runnning the example from galaxy and from RPM is bit different so please follow relevant part.
+
+### Ansible Galaxy
+
+I suppose you followed the part how to install the roles from galaxy, so you roles are already installed in `/etc/ansible/roles/ovirt.ovirt-ansible-roles/roles/` and you have tweaked the `ansible.cfg` to look for those roles.
+
+Now we will run a role called `ovirt-image-template` to download the qcow2 image and create a template from it.
+
+Copy the `ovirt-image-template` role example, I will use `/tmp` as working directory for this example:
+
+```bash
+$ cd /tmp
+$ cp /etc/ansible/roles/ovirt.ovirt-ansible-roles/examples/ovirt_image_template.yml .
+```
+
+### oVirt Ansible Roles RPM
+I suppose you followed the part how to install the roles from RPM, so you roles are already installed in `/usr/share/ansible/roles`.
+
+Now we will run a role called `ovirt-image-template` to download the qcow2 image and create a template from it.
+
+Copy the `ovirt-image-template` role example, I will use `/tmp` as working directory for this example:
+
+```bash
+$ cd /tmp
+$ cp /usr/share/doc/ovirt-ansible-roles/examples/ovirt_image_template.yml .
+```
+
+
+## Tweaking and running the playbook
+
+Modify the variables to reflect your needs:
+
+```yaml
+---
+- name: oVirt image template
+  hosts: localhost
+  connection: local
+  gather_facts: false
+
+  vars_files:
+    # Contains encrypted `engine_password` varibale using ansible-vault
+    - passwords.yml
+
+  vars:
+    engine_url: https://ovirt-engine.example.com/ovirt-engine/api
+    engine_user: admin@internal
+    engine_cafile: /etc/pki/ovirt-engine/ca.pem
+
+    qcow_url: https://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud.qcow2
+    template_cluster: testcluster
+    template_name: centos7_template
+    template_memory: 2GiB
+    template_cpu: 2
+    template_disk_size: 10GiB
+    template_disk_storage: nfs
+
+  roles:
+    - ovirt-image-template
+```
+
+Create a file with oVirt engine user password:
+
+```bash
+$ cat passwords.yml
+engine_password: youruserpassword
+```
+
+Encrypt the file with oVirt user password, it will ask you for a for a value password:
```

Comment:
Suggest:
"Encrypt the file with an oVirt user password. You will be asked for a vault password:"